File: src/MCEncontro1/Exercicio2.java
package MCEncontro1;

import java.util.Scanner;

//Aluno: FELIPE CELESTINO MUROS
/*
Considere um vetor com 50 números inteiros lidos pelo teclado. Faça um algoritmo que imprima o segundo
maior número presente no vetor. Crie e utilize dois procedimentos: uma para preencher o vetor e outro
para imprimir o segundo maior número.
Obs. 1: Não é permitido utilizar qualquer estrutura de dados auxiliar.
Obs. 2: Não é permitido ordenar o vetor.
 */

public class Exercicio2 {

    public static void main(String[] args) {
        int VECTOR_SIZE = 10;        
        int[] vector = new int[VECTOR_SIZE];
        Exercicio2 exercicio = new Exercicio2();

        exercicio.fillVet(vector);
        exercicio.print(vector);
    }

    public Exercicio2() {
    }

    public void fillVet(int[] vector) {
        Scanner scanner = new Scanner(System.in);
        for (int i = 0; i < vector.length; i++) {
            System.out.println("Digite um valor para o vetor na posição " + (i + 1));
            vector[i] = scanner.nextInt();
        }
        scanner.close();
    }

    public void print(int[] vector) {
        int secondBiggest = 0;
        int biggest = 0;
        for (int i = 0; i < vector.length; i++) {
            if (vector[i] > biggest) {
                secondBiggest = biggest;
                biggest = vector[i];
            } else if (vector[i] > secondBiggest) {
                secondBiggest = vector[i];
            }
        }
        System.out.println("O segundo maior número do vetor é " + secondBiggest);    
    }

}

File: src/MCEncontro1/exercicio2.c

//Aluno: FELIPE CELESTINO MUROS
/*
Considere um vetor com 50 números inteiros lidos pelo teclado. Faça um algoritmo que imprima o segundo
maior número presente no vetor. Crie e utilize dois procedimentos: uma para preencher o vetor e outro
para imprimir o segundo maior número.
Obs. 1: Não é permitido utilizar qualquer estrutura de dados auxiliar.
Obs. 2: Não é permitido ordenar o vetor.
 */

#include <stdio.h>

void fillVet(int *vet, int TAM)
{
    for (int i = 0; i < TAM; i++)
    {
        printf("Digite o valor da posicao %d: ", i);
        scanf("%d", &vet[i]);
    }
}

void printSecondBigger(int vet[], int TAM)
{
    int bigger = vet[0], secondBigger = vet[0];
    for (int i = 0; i < TAM; i++)
    {
        if (vet[i] > bigger)
        {
            secondBigger = bigger;
            bigger = vet[i];
        }
        else if (vet[i] > secondBigger && vet[i] != bigger)
        {
            secondBigger = vet[i];
        }
    }
    printf("\nO segundo maior numero eh: %d", secondBigger);
}

int main()
{
    int TAM = 50;
    int vet[TAM];
    fillVet(vet, TAM);
    printSecondBigger(vet, TAM);
    return 0;
}